probe/kubernetes/client.go

```diff
@@ -56,14 +58,39 @@ type client struct {
 }
 
 // runReflectorUntil is equivalent to cache.Reflector.RunUntil, but it also logs
-// errors, which cache.Reflector.RunUntil simply ignores
+// errors (which cache.Reflector.RunUntil simply ignores) and backs off exponentially
+// on errors.
 func runReflectorUntil(r *cache.Reflector, resyncPeriod time.Duration, stopCh <-chan struct{}) {
-	loggingListAndWatch := func() {
-		if err := r.ListAndWatch(stopCh); err != nil {
+	select {
+	case <-stopCh:
+		return
+	default:
+	}
+
+	var err error
+	wait := resyncPeriod
+	for {
+		func() {
+			defer runtime.HandleCrash()
```

Comment:
The `vendor/`ed version doesn't have this comment yet and I presume it only concerns apimachinery since it has restarts in place. But I've heard we are okay if scope crashes so I removed it, thanks!